modules/core/src/effects/lighting-effect.js

```diff
@@ -0,0 +1,53 @@
+import {PointLight as BasePointLight} from 'luma.gl';
+import Effect from '../experimental/lib/effect';
+import {projectPosition} from '../shaderlib/project/project-functions';
+
+// Class to manage ambient, point and directional light sources in deck
+export default class LightingEffect extends Effect {
+  constructor(props) {
+    super(props);
+    this.ambientLight = null;
+    this.directionalLights = [];
+    this.pointLights = [];
+
+    for (const key in props) {
+      const lightSource = props[key];
+
+      switch (lightSource.constructor.name) {
+        case 'AmbientLight':
+          this.ambientLight = lightSource;
+          break;
+
+        case 'DirectionalLight':
+          this.directionalLights.push(lightSource);
+          break;
+
+        case 'PointLight':
+          this.pointLights.push(lightSource);
+          break;
+        default:
+      }
+    }
+  }
+
+  // Pre-project point light positions
+  getProjectedPointLights(viewport) {
+    const projectedPointLights = [];
+    for (let i = 0; i < this.pointLights.length; i++) {
+      const pointLight = this.pointLights[i];
+      const position = projectPosition(pointLight.position, {
+        viewport,
+        coordinateSystem: pointLight.coordinateSystem
```

Comment:
You need to differentiate `fromCoordinateSystem` (aka the coordinate system of the light positions) from `coordinateSystem` (aka the coordinate system of the layer). See https://github.com/uber/deck.gl/blob/master/modules/core/src/shaderlib/project/project-functions.js#L92

Reply:
There is a normalize function in the projectPosition function which set up the fromCoordinateSystem automatically, if we set it manually, need add another tricky code to handle LNGLAT_AUTO_OFFSET_ZOOM_THRESHOLD

Reply:
That's not how it works. You can check the old lighting module code.